specs/Menu.md

```diff
@@ -1,731 +1,672 @@
+
+```typescript
+// v0 shorthand children render callback
+const items = [
+  {
+    key: 'editorials',
+    children: (El, props) => <El>{props.key}</El>
+  },
+]
+
+<Menu defaultActiveIndex={0} items={items} />
+
+// v0 children API custom render
+<Menu>
+  <Menu.Item index={0}>
+    <Menu.ItemContent>Editorials</Menu.ItemContent>
+  </Menu.Item>
+  <Menu.Item index={1}>
+    CustomContent
+  </Menu.Item>
+  {/*Not recommended but definitely possible*/}
+  <div>custom item</div>
+</Menu>
 ```
 
-#### Submenu
+## Variants
 
-Maybe should be named SubmenuContext?
-Should submenu have its own context?
+### Nested menus
 
-```TSX
-function() {
-return (
-    <OpenableContext.Provider value={submenuState}>
-        {props.children}
-    </OpenableContext.Provider>);
-}
-```
+A `Menu` should be able to trigger an additional instance of itself as a part of one or more of its options. The nested `Menu` component should have the same functional capabilities as the root `Menu` component.
 
-#### SubmenuList
+The actions that trigger the the nested `Menu` should be be consistent with the actions that can trigger any root `Menu` from a similar UI control.
 
-```TSX
-function() {
-    const openableContext = React.useContext(OpenableContext);
-    return (openableContext.open &&
-    <Slots.Root {...menuProps}>
-        {props.children}
-    </Slots.Root>);
-}
-```
+We advise that no more than two nested `Menu` components be used, but this spec does not functionally apply that constrain to the implementation of the `Menu` component.
 
-#### SubmenuItem
+### Selection state
+
+A `Menu` should be able to track and represent the selection state of all or some of its options if required.
+
+When an options is associated with a selection state. The `Menu`, either root or nested, should control its dismiss behavior accordingly based on configuration.
+
+### Sections
+
+A `Menu` can be partitioned into sections using visible dividers in its list of options. Each section can contain a heading title that announces or briefly describes the options in the particular section
+
+### Secondary text
+
+An option of a `Menu` component should be able to declare additional secondary text that can provide additional context describing the option or its usage.
+
+For example a secondary text can be a label that shows a keyboard shortcut that will perform an equivalent action of the option of the `Menu` component.
+
+### Split option with nesting
+
+An option of a `Menu` component can trigger a nested `Menu` component and also perform its default action by splitting the option into two interactable areas that handle each action separately.
+
+### Disabled option(s)
+
+All options in a `Menu` component can be disabled and should provide a visible indication for this purpose. User interaction should be defined for disabled options
+
+### Scrollable
```

Comment:
Do we need to support application-provided custom JS scrollbars?

Reply:
I'm pretty sure we do, we need to support hover actions for a scrollbar which native browsers generally do really badly. This is also smth that @assuncaocharles has looked into already for v0, would love to get his experiences